R/ForecastRegrTask.R

```diff
@@ -0,0 +1,58 @@
+#' @title Create a task for univariate forecasting
+#'
+#' @rdname Task
+#'
+#' @description Creates a task for univariate forecasting learners
+#'
+#' @export
+makeForecastRegrTask = function(id = deparse(substitute(data)), data, target,
+  weights = NULL, blocking = NULL, frequency = 1L, fixup.data = "warn",
+  check.data = TRUE) {
+  assertString(id)
+  assertClass(data, "xts")
```

Comment:
> ForecastRegrTask is the only task that takes data not in the form of a data.frame?

Yes and I understand that is something I should clear up and justify.

With time series we really really care about dates. I wanted to ensure we had a strong standard so users would have to do the least amount of work while also making things easy for us. The xts package
1. Enforces POSIXct standardized dates
2. Keeps the data in descending order
3. Ensures the user does not have duplicate dates

When the user bring in data from an xts object we get a very nice and consistent structure. The xts package is also the defacto standard for working with time series and has a lot of nice functions for working with them.  I agree that it would be nice for the user to bring in a data.frame like everything else, but I think that would require us to be explicit with these things while an xts object handles everything for both us and the user.

Now, having said all of that, we could have the user bring in a data.frame and have an argument for `dates`. We could then check the dates for 

1. Uniqueness
2. POSIXct standards
3. Order

Where for (3) if they are out of descending order we sort both the dates and data.frame so that they are compliant and throw a warning to the user that their data was out of order. Then I wonder where do we put these dates? Are they stored in the task? Do we attach them to the data?

I actually like the data.frame approach more, but it's not clear to me the best way to do this. Do you have any suggestions?